ACPs/108-evm-event-importing/README.md

```diff
@@ -0,0 +1,117 @@
+```text
+ACP: 108
+Title: EVM Event Importing Standard
+Author(s): Michael Kaplan (https://github.com/mkaplan13)
+Discussions-To: https://github.com/avalanche-foundation/ACPs/discussions/109
+Status: Proposed
+Track: Best Practices Track
+```
+
+## Abstract
+
+Defines a standard smart contract interface and abstract implementation for importing EVM events from any blockchain within Avalanche using [Avalanche Warp Messaging](https://docs.avax.network/build/cross-chain/awm/overview).
+
+## Motivation
+
+The implementation of Avalanche Warp Messaging within `coreth` and `subnet-evm` exposes a [mechanism for getting authenticated hashes of blocks](https://github.com/ava-labs/subnet-evm/blob/master/contracts/contracts/interfaces/IWarpMessenger.sol#L43) that have been accepted on blockchains within Avalanche. There is currently no clear standard for using authenticated block hashes in smart contracts within Avalanche, making it difficult to build applications that leverage this mechanism. In order to make effective use of authenticated block hashes, contracts must be provided encoded block headers that match the authenticated block hashes and also Merkle proofs that are verified against the state or receipts root contained in the block header. 
+
+With a standard interface and abstract contract implemetation that handles the authentication of block hashes and verification of Merkle proofs, smart contract developers on Avalanche will be able to much more easily create applications that leverage data from other Avalanche blockchains. These type of cross-chain application do not require any direct interaction on the source chain.
+
+## Specification
+
+### Event Importing Interface
+
+We propose that smart contracts importing EVM events emitted by other blockchains within Avalanche implement the following interface.
+
+#### Methods
+
+Imports the EVM event uniquely identified by the source blockchain ID, block header, transaction index, and log index. The event data is provided in the `receiptProof`, which must be a valid Merkle proof of the receipt for the given `txIndex` against the receipt root of the `blockHeader`. Must emit an `EventImported` event upon success. Should validate that the `blockHeader` matches an authenticated block hash from the `sourceBlockchainID`.
+```solidity
+function importEvent(
+    bytes32 sourceBlockchainID,
+    bytes calldata blockHeader,
+    uint256 txIndex,
```

Comment:
nit:  not sure if it feels cleaner with `txIndex` after `receiptProof`

Reply:
I considered it. My rationale for putting `txIndex` first is that the RLP encoded `txIndex` is the key in the receipts trie that the `receiptProof` proves inclusion of (that's obviously not clear from the interface itself though).

Reply:
I added an explanation of this in the section above now. Let me know if you still think it should be re-ordered